### DISCUSSION TITLE
finding character  portraits on tile viewer

### DISCUSSION THREAD

**Discussion by demondzeta** [city41/kof94te#146] [Q&A]
Hi, sorry to bother you. Like I told you before I'm studying your work so I can learn to hack a KOF game. When I followed your tutorial here https://mattgreer.dev/blog/neo-geo-rom-hacking-guide-part-2/ explains a way to track a tile in the tile viewer with the spriteBoundingBoxes.lua script, I then tried to do the same with a KOF game and used the script version you had for that tutorial to track a tile, kof games have 8 C files an d I got this number from the script 78e39 
so we have 4 sets in all
and each one contains  131072 tiles
given that 78e39  is an hex number if we pass it to dec = 495161
the first set then c1 and c2 have tiles from 0 to 131072
c3 and c4 have from 131073 tiles to 262144 
c5 and c6 from 262145 to 393216 
c7 and v8 from 393217 to 524288 and tile number 495161 would be contained here, in the tile viewer though if we load this pair of c files we are not going to find the number 495161 as it will count from 393216 as if it is 0
then to find tile number 495161  we have to make this sustraction 495161  - 393216, I went to this direction and i found, nothing, there was nothing there that looked like the portrait I was looking for, I tried to find then one that you had worked with and i found this file : findings/characterPortraits.md, where you had terry's portrait tile id : 1f716 = 128790 which can be found inside the first C file  set:
![image](https://github.com/user-attachments/assets/e1b9e32c-5aa3-4839-9dde-2c06bd2a15a6)
i went through the same process and found again, nothing : 
![image](https://github.com/user-attachments/assets/ceee3924-44e4-4940-b1a8-e80773fe7d50)
Am I doing something wrong?, are the KOF portraits located inside other files? do I have to do something different or is there a different procedure to find tilees on KOF games, I know this isn't directly an issue with your code but I'll really appreciate if you could help me with this or provide me with a lead on how to solve this, also if there's any way to contact you that you rather prefer instead of this for someone as myself who's following your book and tuts, please let me know

**Comment by city41** [comment]
Terry's portrait at 1f716 is specific to KOF94. It will be in different locations in different KOF games.

I just located his portrait in the KOF94 tiles. I launched kof94 with the spriteBoundingBoxes script, and paused once his portrait was on screen

![image](https://github.com/user-attachments/assets/9e2f6441-0b3a-4072-9571-fac5c75eae73)

In the terminal is this

```bash
Sprite: 221 at (8,520,488), 2 tiles tall
tiles
  1f716
  1f717
palettes
  18
  18
```

1f716 is 128790 in decimal. KOF94's C ROM files have 32,768 tiles in each of them. Not sure where you got 131,072. To figure out which C ROM pair the tile is in, I do `128790 / 32768` which is `3.9303...`, so it's towards the back of the 4th pair.

When I load up the fourth pair (C7/C8) in the tile viewer, it starts at 0 and goes to 32,768. So to find the index in the viewer, it's `128790 % 32768` (modulus operator) which is 30486

![image](https://github.com/user-attachments/assets/ab90fad3-b65c-4cc4-bf49-44447631b6a7)

and that is one of the tile's that makes up Terry's portrait.
